haystack/document_store/base.py

```diff
@@ -65,6 +66,20 @@ def get_all_documents(
         """
         pass
 
+    def __iter__(self):
+        if not self.ids_iterator:
+            self.ids_iterator = [x.id for x in self.get_all_documents()]
+        return self
+
+    def __next__(self):
+        if len(self.ids_iterator) == 0:
+            raise StopIteration
+        else:
+            curr_id = self.ids_iterator[0]
```

Comment:
`curr_id = self.ids_iterator.pop(0)` could be used here so that `self.ids_iterator = self.ids_iterator[1:]`is not necessary.

Reply:
Yup good idea!